agent/agent-profiler/agent-service-profiler/src/main/java/com/microsoft/applicationinsights/serviceprofilerapi/profiler/JfrProfiler.java

```diff
@@ -133,13 +133,23 @@ public boolean initialize(
         flightRecorderConnection = FlightRecorderDiagnosticCommandConnection.connect(mbeanServer);
       }
     } catch (Exception e) {
-      LOGGER.error("Failed to connect to mbean", e);
+      String message =
+          "Unable to initialize the profiler."
+              + (isOpenJ9Jvm()
+                  ? " OpenJ9 JVM is not supported. Instead, please use an OpenJDK JVM."
+                  : "");
+      LOGGER.error(message, e);
       return false;
     }
 
     return true;
   }
 
+  private static boolean isOpenJ9Jvm() {
+    String jvmName = System.getProperty("java.vm.name");
+    return jvmName.contains("OpenJ9");
```

Comment:
```suggestion
    return jvm != null && jvmName.contains("OpenJ9");
```

Reply:
Integrated, not sure jvmName could be null